### PR TITLE
`use-waypoint` is user intent, so should be label

### PR DESCRIPTION
**Pull request by bleggett**
**Please provide a description of this PR:**

This changes `istio.io/use-waypoint`  to be a label, and not an annotation.

Why? 

1. It's not consistent with the other ambient labels, which feels weird:
```
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    istio.io/use-waypoint: waypoint
  creationTimestamp: "2024-04-04T18:13:55Z"
  labels:
    istio.io/dataplane-mode: ambient
    kubernetes.io/metadata.name: default
```

vs

```
apiVersion: v1
kind: Namespace
metadata:
  creationTimestamp: "2024-04-04T18:13:55Z"
  labels:
    istio.io/use-waypoint: waypoint
    istio.io/dataplane-mode: ambient
    kubernetes.io/metadata.name: default
```

2. In general, following a maxim of "labels are for user intent, annotations are for status" gets more consistent results.

3. We will probably want to query this (since it is user intent), and labels are better for queries.

4. If we don't change it now, we never can, so change it now.

Actual change is trivial, most of the diff is test rigging (which is weird TBH)

### CODE REVIEW

**Review by costinm — APPROVED** [review]
Better to use label - but I have few questions:

1. If we have any doc with all the settings around ambient - it probably needs to be updated. I haven't found it yet. 
2. Is "use-waypoint: waypoint" a bit redundant ? use-waypoint: true or "istio.io/waypoint:enabled" or "istio.io/waypoint: default" - if we want later to have 
other modes of waypoints like 'cluster' or 'external' ? 
3. not sure I understand the comments around "defaultNamespace" and if we expect a per-workload label as well if only some workloads use waypoints
4. is it settled to have a single waypoint per namespace ( and I assume have the API limited to boolean and same namespace ) ?

**Review by howardjohn — APPROVED** [review]

**Comment by bleggett** [comment]
> Better to use label - but I have few questions:
> 
>     1. If we have any doc with all the settings around ambient - it probably needs to be updated. I haven't found it yet.

We do not - right now the settings are scattered among various guides.

>     2. Is "use-waypoint: waypoint" a bit redundant ? use-waypoint: true or "istio.io/waypoint:enabled" or "istio.io/waypoint: default" - if we want later to have
>        other modes of waypoints like 'cluster' or 'external' ?

It's `use-waypoint: <name-of-waypoint>`, `waypoint` is just the name used in the example.

>     3. not sure I understand the comments around "defaultNamespace" and if we expect a per-workload label as well if only some workloads use waypoints
> 
>     4. is it settled to have a single waypoint per namespace ( and I assume have the API limited to boolean and same namespace ) ?

Right now `use-waypoint` is the only way to specify a waypoint for an entire namespace, and it does not at this time take an array of values, so yes ATM it is settled, I'd say.

File: pkg/config/constants/constants.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ const (
 	// AmbientRedirectionDisabled is an opt-out, configured by user.
 	AmbientRedirectionDisabled = "disabled"
 
-	// AmbientUseWaypoint is the annotation used to specify which waypoint should be used for a given pod, service, etc...
+	// AmbientUseWaypoint is the label used to specify which waypoint should be used for a given pod, service, etc...
 	AmbientUseWaypoint = "istio.io/use-waypoint"
```
Comment (keithmattix) [inline pkg/config/constants/constants.go:201]: nit: Consider adding Label to the name for clarity (no strongly attached to it though)

**Review by keithmattix — APPROVED** [review]

File: pkg/config/constants/constants.go (hunk 1 above)
Comment (bleggett) [reply pkg/config/constants/constants.go:201]: I kinda wish we did this everywhere and we don't. Might do a followup.

**Comment by bleggett** [comment]
/test integ-basic-arm64

being weird again

**Comment by bleggett** [comment]
cc @hanxiaop or @therealmitchconnors for a UX pass

**Pull request merged**